src/elements/Chapters.tsx

```diff
@@ -1,23 +1,41 @@
 import { Listbox, ListboxItem } from "@nextui-org/react";
+import { useQuery } from "@tanstack/react-query";
+import { getComicChapters } from "../utilities/api";
+import { getRefinedComicChapters } from "../utilities/transformAPIdata";
 
-function Chapters({ latestChapter }: { latestChapter: number }) {
-  const chapters = Array.from(
-    { length: latestChapter + 1 },
-    (_, index) => latestChapter - index
-  );
+function Chapters({
+  latestChapter,
+  comicHid,
+}: {
+  latestChapter: number;
+  comicHid: string;
+}) {
+  const { isPending, isLoading, isError, data } = useQuery({
+    queryKey: ["chaptersArray", comicHid],
+    queryFn: () => getComicChapters(comicHid),
+  });
 
-  return (
-    <>
-      <h1>Chapters</h1>
-      <div className="w-full border-small px-1 py-2 border-default-200 dark:border-default-100">
-        <Listbox onAction={(key) => alert(key)}>
-          {chapters.map((chapter) => (
-            <ListboxItem key={chapter}>Chapter {chapter}</ListboxItem>
-          ))}
-        </Listbox>
-      </div>
-    </>
-  );
+  if (isPending || isLoading) return <h1>Loading....</h1>;
+  if (isError) return <h1>Opps! Error loading data!</h1>;
+
+  if (data) {
+    const comicChapters = getRefinedComicChapters(latestChapter, data);
+
+    return (
+      <>
+        <h1>Chapters</h1>
+        <div className="w-full border-small px-1 py-2 border-default-200 dark:border-default-100">
+          <Listbox onAction={(key) => alert(key)}>
+            {comicChapters.map((chapter) => (
+              <ListboxItem key={chapter.hid}>
+                Chapter {chapter.chap}
+              </ListboxItem>
+            ))}
+          </Listbox>
+        </div>
+      </>
+    );
+  }
```

Comment:
Maybe add some additional safety/error checks? Even if data exists, some values like the hid and chap could be missing. You could check this when calling the `getRefinedComicChapters` and  also confirming that `comicChapters` is not null and has the values you need.

Reply:
Okay, yes I get that. Thank you. Noted, I think i will do that in another PR that is more focused on actually displaying the fetched data in the components.